### ISSUE TITLE
keyboard.wait() causes an error since the last package update

### ISSUE THREAD

**Issue by bobonthenet** [boppreh/keyboard#80]
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Users\rbedard\AppData\Local\Programs\Python\Python36-32\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\Users\rbedard\AppData\Local\Programs\Python\Python36-32\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\rbedard\AppData\Local\Programs\Python\Python36-32\lib\site-packages\keyboard\__init__.py", line 138, in listen
    _os_keyboard.listen(self.queue, _key_table.is_allowed)
  File "C:\Users\rbedard\AppData\Local\Programs\Python\Python36-32\lib\site-packages\keyboard\_winkeyboard.py", line 470, in listen
    prepare_intercept(lambda e: queue.put(e) or is_allowed(e.name, e.event_type == KEY_UP))
  File "C:\Users\rbedard\AppData\Local\Programs\Python\Python36-32\lib\site-packages\keyboard\_winkeyboard.py", line 451, in prepare_intercept
    keyboard_hook = SetWindowsHookEx(WH_KEYBOARD_LL, keyboard_callback, NULL, NULL)
ctypes.ArgumentError: argument 2: <class 'TypeError'>: expected CFunctionType instance instead of CFunctionType
```

I'm doing the following and I get the above exception when I hit the keyboard.wait line. I did not have a chance to dig in to figure out what was happening so I'm hoping someone can get to this before I am able to. This was working fine before I updated the keyboard package today.

```
recorded = []
keyboard.hook(recorded.append)
mouse.hook(recorded.append)
keyboard.wait('esc')
```

**Comment by boppreh** [comment]
Checking it.

**Comment by boppreh** [comment]
I couldn't reproduce the issue in either Python3.6 64 bits or Python2.7 32 bits, and the error makes little sense ("expected CFunctionType instance instead of CFunctionType" ?!). But since those lines are just type-checking, I can remove them without breaking anything.

Can you check if the latest version in this repo (https://github.com/boppreh/keyboard/tree/e8124699f9582c2970251c0bc442f0717221a3b0) solves the problem?

**Comment by bobonthenet** [comment]
I get the same error with the latest version
Windows 7 64 bit
Python 3.6.1 32 bit

**Comment by bobonthenet** [comment]
The program hangs after the error. When I kill it with ctrl+c I get the following.

```
Traceback (most recent call last):
  File "C:/Users/rbedard/Desktop/test.py", line 7, in <module>
    keyboard.wait('esc')
  File "C:\Users\rbedard\AppData\Local\Programs\Python\Python36-32\lib\site-packages\keyboard\__init__.py", line 656, in wait
    wait()
  File "C:\Users\rbedard\AppData\Local\Programs\Python\Python36-32\lib\site-packages\keyboard\__init__.py", line 643, in wait
    return q.get(timeout=1)
KeyboardInterrupt
```

I can't think of anything on my pc that has changed other than the keyboard package.

**Comment by boppreh** [comment]
I'm sorry to ask, but just to make sure: did you use the master branch from Github (`git clone https://github.com/boppreh/keyboard`) or one of the releases? The change is only available in the master branch at the moment.

The hanging is expected after that error. It failed to setup the keyboard hook, so it can't detect any keypresses, so `wait` never finishes by itself.

**Comment by bobonthenet** [comment]
I am using the master branch.

**Comment by bobonthenet** [comment]
So I'm actually thinking something changed on my system. I just tried my earlier fork which I know for certain worked and I get the same error. Now I just need to figure out what would have changed.

**Comment by bobonthenet** [comment]
I just removed Python completely and installed 64 bit. Same error.

**Comment by boppreh** [comment]
At least it's consistent, but I still can't reproduce. I'll go read some more docs about those win32 functions and ctypes, see if there are hints. But I'm completely clueless at the moment, sorry.

**Comment by bobonthenet** [comment]
Well, I got home where I can actually spend time digging into this and of course, I can't replicate it on my home PC.

**Comment by bobonthenet** [comment]
Nope sorry spoke too soon. I had keyboard-0.9.13 still, keyboard-0.10.0 is broken :(
I am going to grab a bite to eat then I should be able to dig in and find out what is going on.

**Comment by bobonthenet** [comment]
I was mistaken before. The error is happening on my keyboard.hook(recorded.append) line. The failure is happing in _generic.py line 41 `self.listening_thread.start()`

**Comment by bobonthenet** [comment]
I still don't know what the problem is but it looks like it was introduced with d24e3eb. The commit prior to that does not give me the error. I haven't had the time to work on this that I'd hoped.

**Comment by boppreh** [comment]
That's incredibly helpful, thank you. I'm diagnosing the problem right now.

**Comment by bobonthenet** [comment]
You're welcome. I'm committed to helping you resolve this because I've found this library to be incredibly useful. I'm working on my own utility that leans very heavily on this functionality.

**Comment by boppreh** [comment]
Still trying to find the problem. Makes no sense since the last commit that remove the type checking. Can you check if the error is different, at least?

**Comment by bobonthenet** [comment]
I got it. I don't understand enough to know why this works but I also removed the type checking from _winmouse.py and that cleared up the error. This was especially confusing because - to answer your last question - the error was not different and never mentions the _winmouse.py file but does call out the _winkeyboard.py file where you removed the type checking. Also, doing `python -m keyboard` does not produce the error. Probably because there was no error there to trigger. I wrote this to help me with testing.

```
import keyboard
from keyboard import mouse
recorded = []
keyboard.hook(recorded.append)
mouse.hook(recorded.append)
keyboard.wait('esc')
keyboard.unhook(recorded.append)
mouse.unhook(recorded.append)
```

**Comment by bobonthenet** [comment]
I submitted a new pull request to comment out the type checking in _winmouse.py

**Comment by bobonthenet** [comment]
I'm still curious as to why this fixes the issue. I read through some of the ctypes documentation but it is over my head right now. If you can give me an ELI5 explanation I'd really appreciate it. I feel like this is enough to close the bug for now though. Can you update the package too please?

**Comment by boppreh** [comment]
Aahh, so the error was in the mouse part. That sheds some light. Makes sense that `python -m keyboard` didn't trigger the error, since it doesn't hook to any of the mouse parts. `python -m keyboard.mouse` should have been used then.

I've already approved your pull request. Thank you very much for that. I'm still not sure why the ctypes type-checking is failing, but since it's done at runtime it doesn't add much anyway.

Also note that I'm in the process of splitting the `mouse` parts to a [separate package](https://github.com/boppreh/mouse). I've already applied your fix there and [pushed a new release](https://pypi.python.org/pypi/mouse). Which reminds me to update the README here...

Thank you for reporting, diagnosing and fixing the problem yourself. Sorry I could not be of more help, but I'm glad the issue is fixed.

**Comment by boppreh** [comment]
> I'm still curious as to why this fixes the issue. I read through some of the ctypes documentation but it is over my head right now. If you can give me an ELI5 explanation I'd really appreciate it. 

`ctypes` is for Python code to communicate with environments that expect C, such was Windows' API win32. There are a lot of differences between C and Python, such as features present in one side but not in the other. This stdlib module offers functions like `byref(obj)` that behave like C references, and primitive types like `c_int`, to convert Python numbers to stuff that C expects.

Manually converting objects is boring, and any mistakes will likely lead to segfaults or worse memory corruption. So `ctypes` allows you to annotate the Python objects that represent C functions, with the expected arguments and return types ([example](https://github.com/boppreh/keyboard/blob/f1d10936c507f141f80e72333d2b0d4f22f48f73/keyboard/_winkeyboard.py#L84)). So you can declare that some existing C function takes an integer and a pointer to a function that takes a long and a pointer to a buffer... and so on.

In this issue, it seems that one of the arguments of `SetWindowsHookEx` was declared, in Python, to be another function, with certain arguments and return types (`LowLevelMouseProc`). But what was actually passed was a function with different arguments and return type.s It's hard to know more details because it just reports as `CFunctionType`. This is most likely an issue in the type-checking itself, and not the type-checked code.

I still don't understand why I can't reproduce this issue, since I'm also using 64-bit Windows.

> Can you update the package too please?

v0.10.3 should have your fix. Also, mouse feature are being migrated to the [`mouse`[(https://github.com/boppreh/mouse) (`pip install mouse`). I've already ported your fix there.
